src/Common/src/Interop/Unix/System.Native/Interop.MountPoints.cs

```diff
@@ -0,0 +1,303 @@
+            internal long                   TotalFreeSpace;
+            internal long                   TotalSize;
+        }
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private unsafe delegate void MountPointFound(byte* name, int nameLength);
+
+        [DllImport(Libraries.SystemNative)]
+        private static extern int GetAllMountPoints(MountPointFound mpf);
+
+        [DllImport(Libraries.SystemNative)]
+        internal static extern int GetSpaceInfoForMountPoint([MarshalAs(UnmanagedType.LPStr)]string name, out MountPointInformation mpi);
+
+        [DllImport(Libraries.SystemNative)]
+        private unsafe static extern int GetFormatInfoForMountPoint(
+            [MarshalAs(UnmanagedType.LPStr)]string name,
+            byte* formatNameBuffer,
+            int bufferLength,
+            long* formatType);
+
+        internal static int GetFormatInfoForMountPoint(string name, out string format)
+        {
+            DriveType temp;
+            return GetFormatInfoForMountPoint(name, out format, out temp);
+        }
+
+        internal static int GetFormatInfoForMountPoint(string name, out DriveType type)
+        {
+            string temp;
+            return GetFormatInfoForMountPoint(name, out temp, out type);
+        }
+
+        private static int GetFormatInfoForMountPoint(string name, out string format, out DriveType type)
+        {
+            unsafe
+            {
+                byte* formatBuffer = stackalloc byte[MountPointFormatBufferSizeInBytes];    // format names should be small
+                long numericFormat;
+                int result = GetFormatInfoForMountPoint(name, formatBuffer, MountPointFormatBufferSizeInBytes, &numericFormat);
+                if (result == 0)
+                {
+                    // Check if we have an numeric answer or string
+                    if (numericFormat != -1)
+                    {
+                        format = Enum.GetName(typeof(LinuxFileSystemTypes), numericFormat);
+                        type = GetDriveType(format);
+                    }
+                    else
+                    {
+                        format = Marshal.PtrToStringAnsi((IntPtr)formatBuffer);
+                        type = GetDriveType(format);
+                    }
+                }
+                else
+                {
+                    format = string.Empty;
+                    type = DriveType.Unknown;
+                }
+
+                return result;
+            }
+        }
+
+        internal static string[] GetAllMountPoints()
+        {
+            List<string> lst = new List<string>();
+            unsafe
+            {
+                int result = GetAllMountPoints((byte* name, int nameLength) =>
+                {
+                    lst.Add(Marshal.PtrToStringAnsi((IntPtr)name, nameLength));
+                });
+            }
+
+            return lst.ToArray();
```

Comment:
Good idea, this also makes GetDrives more efficient since we can determine how big of an array to create and avoid a ToArray() call on List<T>